### PR TITLE
STOR-2017: Bump all deps for 4.18

### PR DESCRIPTION
**Pull request by rhrmo**
Bump all deps for 4.18

Notes:
* Updated calls for functions coming from library-go

### CODE REVIEW

File: pkg/csoclients/csoclients.go [hunk 1]
```diff
@@ -127,9 +133,13 @@ func NewClients(controllerConfig *controllercmd.ControllerContext, resync time.D
 	}
 	c.MonitoringInformer = prominformer.NewSharedInformerFactory(c.MonitoringClient, resync)
 
-	c.OperatorClient = &operatorclient.OperatorClient{
-		Informers: c.OperatorInformers,
-		Client:    c.OperatorClientSet,
+	gvr := operatorv1.SchemeGroupVersion.WithResource("storages")
+	gvk := operatorv1.SchemeGroupVersion.WithKind("Storages")
```
[the remark this answers is not in the document]
Comment (rhrmo) [reply pkg/csoclients/csoclients.go:137]: Oh yes, you're right, thank you

File: pkg/csoclients/csoclients.go [hunk 2]
```diff
@@ -286,3 +307,33 @@ func StartMgmtInformers(clients *Clients, stopCh <-chan struct{}) {
...
+	if err != nil {
+		return nil, fmt.Errorf("unable to extract fields for %q: %w", fieldManager, err)
+	}
+	if ret.Spec == nil {
+		return nil, nil
+	}
+	return &ret.Spec.OperatorSpecApplyConfiguration, nil
+}
+func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorStatusApplyConfiguration, error) {
+	castObj := &operatorv1.ClusterCSIDriver{}
```
[the remark this answers is not in the document]
Comment (rhrmo) [reply pkg/csoclients/csoclients.go:326]: Yes it should, thank you!

File: pkg/csoclients/csoclients.go [hunk 3]
```diff
@@ -231,9 +241,18 @@ func NewHypershiftGuestClients(
 	}
 	c.MonitoringInformer = prominformer.NewSharedInformerFactory(c.MonitoringClient, resync)
 
-	c.OperatorClient = &operatorclient.OperatorClient{
-		Informers: c.OperatorInformers,
-		Client:    c.OperatorClientSet,
+	gvr := operatorv1.SchemeGroupVersion.WithResource("storages")
+	gvk := operatorv1.SchemeGroupVersion.WithKind("Storages")
```
Comment (dobsonj) [inline pkg/csoclients/csoclients.go:245]: gvr should be plural (`storages`) but gvk should be singular (`Storage`).
You have it right in NewClients() above, this change could explain why the hypershift e2e job is failing:
```
    eventually.go:220: observed *v1beta1.HostedCluster e2e-clusters-nb7vg/example-76h7n invalid at RV 108746 after 2s: incorrect condition: wanted ClusterVersionProgressing=False, got ClusterVersionProgressing=True: ClusterOperatorDegraded(Working towards 4.18.0-0.ci.test-2024-11-14-151003-ci-op-sic0h13g-latest: 493 of 612 done (80% complete), waiting up to 40 minutes on storage)
```

File: pkg/csoclients/csoclients.go [hunk 4]
```diff
@@ -286,3 +307,33 @@ func StartMgmtInformers(clients *Clients, stopCh <-chan struct{}) {
 		informer.Start(stopCh)
 	}
 }
+
+func extractOperatorSpec(obj *unstructured.Unstructured, fieldManager string) (*applyoperatorv1.OperatorSpecApplyConfiguration, error) {
+	castObj := &operatorv1.Storage{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, castObj); err != nil {
+		return nil, fmt.Errorf("unable to convert to ClusterCSIDriver: %w", err)
```
Comment (dobsonj) [inline pkg/csoclients/csoclients.go:314]: Please update the log message here and in `extractOperatorStatus` to replace `ClusterCSIDriver` with `Storage`.

File: pkg/csoclients/csoclients.go (hunk 3 above)
Comment (rhrmo) [reply pkg/csoclients/csoclients.go:245]: I see, sorry I did not notice, thank you!

**Comment by dobsonj** [comment]
Thanks!
/lgtm
/approve
/label px-approved
/label docs-approved

**Comment by rhrmo** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-e2e-aws-ovn-upgrade

**Comment by rhrmo** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.18-upgrade-from-stable-4.17-e2e-gcp-ovn-rt-upgrade

**Pull request merged**